tests/test_normalizers.py

```diff
@@ -73,6 +73,12 @@ def test_authority_normalization():
     assert uri.authority == 'user%2AName@example.com'
 
 
+def test_ipv6_zone_id_normalization():
+    uri = URIReference(
+        None, '[::1%eth0]', None, None, None, None).normalize()
```

Comment:
Would you be willing to parametrize this with this host and the ones from the builder tests?

Reply:
I believe I've accomplished what you wanted here?